src/WordCloud/WordCloud.tsx

```diff
@@ -1,83 +1,83 @@
 import { ExplanationData } from "./types";
 import WordBounds from "./WordBounds";
-import { Rectangle, Word, WordCloudData } from "./components/types";
+import {
+  PlacedWordCloud,
+  Rectangle,
+  Word,
+  WordCloudData,
+} from "./components/types";
 
-const useWordCloudLayout = (wordClouds: WordCloudData): WordCloudData => {
+const useWordCloudLayout = (wordClouds: WordCloudData): PlacedWordCloud => {
   wordClouds.forEach((cloud, cloudIdx, originalWordClouds) => {
-    // Initialize the weights with the value 1, of the size of the number of intervals
-    const weight = new Array(NUMBER_OF_INTERVALS).fill(1);
-
     const newWordCloud = cloud.words.slice(1).reduce<Word[]>(
       (placedWords, word) => {
-        const futureItem = futurPosition(
+        const futureItem = futureSpiralPosition(
           word.rect,
-          placedWords.map(({ rect }) => rect),
-          3,
-          weight
+          placedWords.map(({ rect }) => rect)
         );
         return [...placedWords, { ...word, rect: futureItem }];
       },
       [cloud.words[0]]
     );
     originalWordClouds[cloudIdx].words = newWordCloud;
-    originalWordClouds[cloudIdx].bound = boundParent(
-      newWordCloud.map(({ rect }) => rect)
-    );
   });
 
-  wordClouds.sort((cloudA, cloudB) => {
-    return cloudA.bound &&
-      cloudB.bound &&
-      cloudA.bound.width * cloudA.bound.height >=
-        cloudB.bound.width * cloudB.bound.height
+  const placedWordCloud: PlacedWordCloud = wordClouds.map((cloud) => ({
+    ...cloud,
+    bound: getBoundingWordCloud(cloud.words),
+  }));
+
+  placedWordCloud.sort((cloudA, cloudB) => {
+    return cloudA.bound.width * cloudA.bound.height >=
+      cloudB.bound.width * cloudB.bound.height
       ? -1
       : 1;
   });
 
-  // Initialize the weights with the value 1, of the size of the number of intervals
-  const weight = new Array(NUMBER_OF_INTERVALS).fill(1);
-  wordClouds.forEach((cloud, cloudIdx, originalWordClouds) => {
-    if (cloudIdx === 0 || !cloud.bound) {
-      return;
-    }
-    const futurePos = futurPosition(
-      cloud.bound,
-      originalWordClouds
-        .slice(0, cloudIdx)
-        .map(({ bound }) => bound as Rectangle),
-      4,
-      weight
-    );
-    const slidedWordRectangles = slideWords(
-      originalWordClouds[cloudIdx].words.map(({ rect }) => rect),
-      getMoveDirection([originalWordClouds[0].bound as Rectangle], futurePos)
-    );
-    originalWordClouds[cloudIdx].words = originalWordClouds[cloudIdx].words.map(
-      (w, idx) => ({
-        ...w,
-        rect: slidedWordRectangles[idx],
-      })
-    );
-    const newBound = boundParent(slidedWordRectangles);
-    console.log(newBound);
-    originalWordClouds[cloudIdx].bound = newBound;
-  });
+  const newWordClouds = placedWordCloud.reduce<PlacedWordCloud>(
+    (accWordClouds, cloud, cloudIdx) => {
+      if (cloudIdx === 0) {
+        accWordClouds.push({
+          ...cloud,
+          bound: {
```

Comment:
Does the new bound resolve the collision issue?

Reply:
Actually, I don't really know what solved the issue, I think there was something weird in the slide words function, also the boundParent was returning a bound using a different coordinate system (center point, width and height) instead of (top left corner, width and height). I also used a different layout technique based on a spiral. We can discuss that later.